src/codegen/irgen/util.cpp

```diff
@@ -92,6 +92,33 @@ llvm::Constant* getStringConstantPtr(const char* str) {
 // to some associated compiler-level data structure.
 // It's slightly easier to emit them as integers (there are primitive integer constants but not pointer constants),
 // but doing it this way makes it clearer what's going on.
+
+static llvm::StringMap<void*> relocatable_syms;
+
+void clearRelocatableSymsMap() {
+    relocatable_syms.clear();
+}
+
+void* getValueOfRelocatableSym(const std::string& str) {
+    auto it = relocatable_syms.find(str);
+    if (it != relocatable_syms.end())
+        return it->second;
+    return NULL;
+}
+
+llvm::Constant* embedRelocatablePtr(const void* addr, llvm::Type* type) {
+    assert(addr);
+
+    if (!ENABLE_JIT_OBJECT_CACHE)
+        return embedConstantPtr(addr, type);
+
+    std::string name = (llvm::Twine("c") + llvm::Twine(relocatable_syms.size())).str();
+    relocatable_syms[name] = (void*)addr;
```

Comment:
gcc (4.8?) doesn't like this :P (`cast from type ‘const void*’ to type ‘void*’ casts away qualifiers`).